src/ring_theory/nullstellensatz.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import ring_theory.jacobson
+import field_theory.algebraic_closure
+import field_theory.mv_polynomial
+
+/-!
+  # Nullstellensatz
+  This file establishes a version of Hilbert's classical Nullstellensatz for `mv_polynomial`s.
+  The main statement of the theorem is `vanishing_ideal_zero_locus_eq_radical`.
+
+  The statement is in terms of new definitions `vanishing_ideal` and `zero_locus`.
+  Mathlib already has versions of these in terms of the prime spectrum of a ring,
+    but those are not well-suited for expressing this result.
+  Suggestions for better ways to state this theorem or organize things are welcome.
+
+  The machinery around `vanishing_ideal` and `zero_locus` is also minimal, I only added lemmas
+    directly needed in this proof, since I'm not sure if they are the right approach.
```

Comment:
I believe markdown will turn lines starting with spaces into `<pre>` blocks (for inline code).

```suggestion
# Nullstellensatz
This file establishes a version of Hilbert's classical Nullstellensatz for `mv_polynomial`s.
The main statement of the theorem is `vanishing_ideal_zero_locus_eq_radical`.

The statement is in terms of new definitions `vanishing_ideal` and `zero_locus`.
Mathlib already has versions of these in terms of the prime spectrum of a ring,
but those are not well-suited for expressing this result.
Suggestions for better ways to state this theorem or organize things are welcome.

The machinery around `vanishing_ideal` and `zero_locus` is also minimal, I only added lemmas
directly needed in this proof, since I'm not sure if they are the right approach.
```